src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java

```diff
@@ -279,35 +314,21 @@ public void onTextAvailable(ProcessEvent event, Key outputType) {
         output.add(event.getText());
       }
     });
-
     runningPantsProcesses.put(currentProject, process);
     processHandler.runProcess();
     runningPantsProcesses.remove(currentProject, process);
-
-    final boolean success = process.exitValue() == 0;
-    // Mark project dirty if compile failed.
-    if (!success) {
-      FileChangeTracker.markDirty(currentProject);
-    }
-    else {
-      FileChangeTracker.addManifestJarIntoSnapshot(currentProject);
-    }
-    notifyCompileResult(success);
-    // Sync files as generated sources may have changed after Pants compile.
-    PantsUtil.synchronizeFiles();
     String finalOutString = String.join("", output);
-    Pair<Boolean, Optional<String>> result = Pair.create(success, Optional.of(finalOutString));
-    return result;
+
+    return new PantsExecuteTaskResult((process.exitValue() == 0),
+                                      Optional.of(finalOutString));
   }
 
-  private void notifyCompileResult(final boolean success) {
-  /* Show pop up notification about pants compile result. */
+  private static void notify(final String title, final String subtitle, NotificationType type) {
+    /* Show pop up notification about pants compile result. */
     ApplicationManager.getApplication().invokeLater(new Runnable() {
       @Override
       public void run() {
-        String message = success ? "Pants compile succeeded." : "Pants compile failed.";
-        NotificationType type = success ? NotificationType.INFORMATION : NotificationType.ERROR;
-        Notification start = new Notification(PantsConstants.PANTS, "Compile message", message, type);
+        Notification start = new Notification(PantsConstants.PANTS, PantsIcons.Icon, title, subtitle, null, type, null);
```

Comment:
any reason for using a different api for `new Notification`?

Reply:
`PantsMakeBeforeRun` uses `ApplicationManager` here to invoke the notification later (which is what it was doing before, I just put that into this private `.notify()` method). I was 
thinking about making `.notify()` public and call that in `PantsTaskActionBase`, which is the other place a notification is created in this diff, but that might cause weirdness with integration tests since `ApplicationManager` is used here, so I left the notification creation code in two different places.